avro-fastserde/src/test/java/com/linkedin/avro/fastserde/FastStringableTest.java

```diff
@@ -160,6 +162,35 @@ public void javaStringPropertyTest(Boolean whetherUseFastDeserializer) throws IO
     }
   }
 
+  @Test(groups = {"deserializationTest"}, dataProvider = "SlowFastDeserializer")
+  public void javaStringPropertyInsideUnionTest(Boolean whetherUseFastDeserializer) {
+    Schema schema = createRecord(
+        createField("name", javaStringSchema),
+        createUnionField("favorite_number", Schema.create(Schema.Type.INT)),
+        createUnionField("favorite_color", javaStringSchema)
+    );
+    GenericRecord record  = new GenericData.Record(schema);
+    record.put("name", "test_user");
+    record.put("favorite_number", 10);
+    record.put("favorite_color", "blue");
+    Decoder decoder = writeWithFastAvro(record, schema, false);
+
+    GenericRecord afterDecoding;
+    if (whetherUseFastDeserializer) {
+      afterDecoding = readWithFastAvro(schema, schema, decoder, false);
+    } else {
+      afterDecoding = readWithSlowAvro(schema, schema, decoder, false);
+    }
+
+    if (Utils.isAvro14()){
```

Comment:
nit - < 1.5 is probably safer. also - do we know which major avro version introduced String support? is it only 1.4 thats missing it?

Reply:
So far, it should be fine since fast-avro only supports 1.4/1.7/1.8, and other avro versions should fail back to the default avro lib.